docs/developer-guide/extensions/proxy-extensions.md

```diff
@@ -0,0 +1,190 @@
+route.
+
+#### `extensions.backend.connectionTimeout` (*duration string*)
+(optional. Default: 2s)
+
+Is the maximum amount of time a dial to the extension server will wait
+for a connect to complete. 
+
+#### `extensions.backend.keepAlive` (*duration string*)
+(optional. Default: 15s)
+
+Specifies the interval between keep-alive probes for an active network
+connection between the API server and the extension server. Optional
+field.
+
+#### `extensions.backend.idleConnectionTimeout` (*duration string*)
+(optional. Default: 60s)
+
+Is the maximum amount of time an idle (keep-alive) connection between
+the API server and the extension server will remain idle before
+closing itself.
+
+#### `extensions.backend.maxIdleConnections` (*int*)
+(optional. Default: 30)
+
+Controls the maximum number of idle (keep-alive) connections between
+the API server and the extension server.
+
+#### `extensions.backend.services` (*list*)
+
+Defines a list with backend url by cluster.
+
+#### `extensions.backend.services.url` (*string*)
+(mandatory)
+
+Is the address where the extension backend must be available.
+
+#### `extensions.backend.services.cluster` (*string*)
+(optional)
+
+If provided, will have to match the application destination name to
+have requests properly forwarded to this service URL.
+
+## Usage
+
+Once a proxy extension is configured it will be made available under
+the `/extensions/<extension-name>` endpoint exposed by Argo CD API
+server. The example above will proxy requests to
+`<apiserver-host>/extensions/httpbin/` to `http://httpbin.org`.
+
+The diagram below illustrates an interaction possible with this
+configuration:
+
+```
+                                              ┌─────────────┐
+                                              │ Argo CD UI  │
+                                              └────┬────────┘
+                                                   │  ▲
+  GET <apiserver-host>/extensions/httpbin/anything │  │ 200 OK
+            + authn/authz headers                  │  │
+                                                   ▼  │
+                                            ┌─────────┴────────┐
+                                            │Argo CD API Server│
+                                            └──────┬───────────┘
+                                                   │  ▲
+                   GET http://httpbin.org/anything │  │ 200 OK
+                                                   │  │
+                                                   ▼  │
+                                             ┌────────┴────────┐
+                                             │Backend Extension│
+                                             └─────────────────┘
+```
+
+Note that Argo CD API Server requires additional HTTP headers to be
+sent in order to enforce if the incoming request is authenticated and
+authorized before being proxied to the Backend Extension. The headers
+are documented below:
```

Comment:
Worth noting that some of these (but not all) are passed along to the backend? For example, I don't think `Cookie` (or any other headers) are forwarded.

Reply:
All headers that are not removed in the `sanitizeRequest` function are forwarded to the backend.
Not sure if this is answering your question?

Reply:
I will add the docs to clarify how the request sanitize works

Reply:
done